File: modules/silent-sms/ios/SilentSmsModule.swift
import ExpoModulesCore

public class SilentSmsModule: Module {
  public func definition() -> ModuleDefinition {
    Name("SilentSms")

    AsyncFunction("isAvailableAsync") { () -> Bool in
      return false
    }

    AsyncFunction("sendSilentSms") { (recipients: [String], message: String) -> Bool in
      // Programmatic background SMS without user interaction is not permitted by iOS platform policy
      return false
    }
  }
}

File: modules/silent-sms/src/SilentSmsModule.ts
import { NativeModule, requireNativeModule } from 'expo';

declare class SilentSmsModule extends NativeModule {
  sendSilentSms(recipients: string[], message: string): Promise<boolean>;
  isAvailableAsync(): Promise<boolean>;
}

export default requireNativeModule<SilentSmsModule>('SilentSms');
